halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/problem/ProblemSet.java

```diff
@@ -50,10 +50,11 @@ public ProblemSet() { }
   /**
    * This will throw an exception if this problem set stores any problems.
    */
-  public void throwIfProblem() {
+  public ProblemSet throwIfProblem() {
```

Comment:
Is this a common pattern? Otherwise why does this return a ProblemSet when add() and sort() dont? If anything I'd expect the other way around.
It seems odd to be returning a problem set since you would not expect these methods to be mutable.